nucliadb_vectors/src/data_point/mod.rs

```diff
@@ -415,6 +420,7 @@ impl<'a, Dlog: DeleteLog> Retriever<'a, Dlog> {
         delete_log: &'a Dlog,
         similarity: Similarity,
         min_score: f32,
+        dimension: usize,
```

Comment:
Well, ideally we would have the dimension inside the index metadata and just pass the metadata around. As it is now, you have to get the dimension from various places, so you would be constructing the struct in every caller. It would be a bit easier to understand (since you see the name of the fields explicitly) but still ugly, I think. Or did you have something else in mind?

Reply:
I just meant grouping `similarity`, `min_score` and `dimension` in one struct since we are passing them together a lot.